src/submission/models.py

```diff
@@ -1015,11 +1018,20 @@ def get_article(journal, identifier_type, identifier):
             # resolve an article from an identifier type and an identifier
             if identifier_type.lower() == 'id':
                 # this is the hardcoded fallback type: using built-in id
-                article = Article.objects.filter(id=identifier, journal=journal)[0]
+                article = Article.objects.filter(
+                    id=identifier,
+                    journal=journal,
+                ).exclude(
+                    stage=STAGE_ARCHIVED,
```

Comment:
doing this via the default manager might be good here too
In this case we would need a custom manager to retrieve the rejrected/archived articles explicitely